src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -2857,6 +2857,383 @@ ep_rt_mono_write_event_ee_startup_start (void)
 		NULL);
 }
 
+// !!!!!!! NOTE !!!!!!!!
+// The flags must match those in the ETW manifest exactly
+// !!!!!!! NOTE !!!!!!!!
+
+typedef enum {
+	K_ETW_TYPE_FLAGS_ARRAY = 0x8,
+} EtwTypeFlags;
+
+// This only contains the fixed-size data at the top of each struct in
+// the bulk type event.  These fields must still match exactly the initial
+// fields of the struct described in the manifest.
+typedef struct _EventStructBulkTypeFixedSizedData {
+	uint64_t type_id;
+    uint64_t module_id;
+    uint32_t type_name_id;
+    uint32_t flags;
+	uint8_t cor_element_type;
+} EventStructBulkTypeFixedSizedData;
+
+// Represents one instance of the Value struct inside a single BulkType event
+typedef struct _BultTypeValue {
+	EventStructBulkTypeFixedSizedData fixed_sized_data;
+	uint32_t c_type_parameters;
+	intptr_t rg_type_parameters[100]; // Should be variable length to save space
+	char *s_name;
```

Comment:
CoreCLR uses a prealloced stack array for this, `StackSArray<ULONGLONG>`.

Reply:
When I first looked at it, I wasn't sure what the size of CoreCLR's `StackSArray<ULONGLONG>` was, and naively thought it was an array that can allocate more space on demand. Now with another take, I see that https://github.com/dotnet/runtime/blob/6af4abc5a0db0560e719539a869233d15d28dc09/src/coreclr/inc/sarray.h#L218 has `STACK_ALLOC/sizeof(ELEMENT)` which in this case would be `STACK_ALLOC/sizeof(uint64_t)` and by https://github.com/dotnet/runtime/blob/dbe06f0ce86455149ab140b383f788f9ec110f96/src/coreclr/inc/sbuffer.h#L557, this would be `256/8`.